solutions/usaco-1011.mdx

```diff
@@ -61,5 +61,35 @@ public class triangles{
 ```
 
 </JavaSection>
+ 
+<PySection>
+
+```py
+file_in = open('triangles.in')
+data = file_in.read().strip().split('\n')
+n = int(data[0])
+
+vertices = [list(map(int, data[i + 1].split())) for i in range(n)]
+max_area = 0
+
+# Brute force through all possible triangles to find the largest possible area
+for i in range(n):
+	for j in range(n):
+		if i != j and vertices[j][1] == vertices[i][1]:
+			for k in range(n):
+				if (k != j and k != i) and vertices[i][0] == vertices[k][0]:
+					area = abs(vertices[k][1] - vertices[i][1]) * abs(vertices[j][0] - vertices[i][0])
+					max_area = max(max_area, area)
+		elif i != j and vertices[j][0] == vertices[i][0]:
+			for k in range(n):
+				if (k != j and k != i) and vertices[i][1] == vertices[k][1]:
+					area = abs(vertices[k][0] - vertices[i][0]) * abs(vertices[j][1] - vertices[i][1])
+					max_area = max(max_area, area)
```

Comment:
Redundant; code passes fine without these lines ( only assuming i and j horizontal )

Reply:
Wait so I should just delete lines 83-87? Will the code still work perfectly?

Reply:
> ( only assuming i and j horizontal )

Reply:
But what if i and j aren't horizontal, then won't the code fail?

Reply:
then loop through all pairs, (i, j) such that they are horizontal (?)

Reply:
But how would i account for the cases when they aren't

Reply:
loop through all pairs and check if they're horizontal...

Reply:
Wait but then what happens if they arent (sry for bothering)

Reply:
Yes you delete the lines and the code will work perfectly.

Reply:
> Wait but then what happens if they arent

Every triangle that you consider has a horizontal edge. This is the (i, j) edge -- it will always exist

Reply:
@AAkashyap1 ^^